pandas/core/arrays/period.py

```diff
@@ -986,13 +985,15 @@ def raise_on_incompatible(left, right) -> IncompatibleFrequency:
     if isinstance(right, (np.ndarray, ABCTimedeltaArray)) or right is None:
         other_freq = None
     elif isinstance(right, BaseOffset):
-        other_freq = freq_to_period_freqstr(right.n, right.name)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
```

Comment:
looks a bit dangerous :) is this for the business day warning? if so, I'd suggest either enforcing that one first, or using `warnings.filterwarnings` and only filtering the warning you mean to

Reply:
yes, this for the business day warning. I agree, it's better to use `warnings.filterwarnings` instead of `warnings.simplefilter`. I made changes in order to filter: `"Future warning "PeriodDtype[B] is deprecated"`

Reply:
we can rid of this `warnings.filterwarnings` after enforcing depreciation of `PeriodDtype[B]`, or we can postpone merging this PR until enforcing deprecation of PeriodDtype[B] is done.